src/main/java/duke/ui/Parser.java

```diff
@@ -0,0 +1,155 @@
+     *
+     * @param userInput User's string input.
+     * @return Command object to be executed.
+     * @throws DukeException
+     */
+    public Command parse(String userInput) throws DukeException {
+        Commands command = Parser.getCommand(userInput); // will catch any invalid command alrdy
+        Command c = null;
+        switch (command) {
+        case BYE:
+            c = new ExitCommand();
+            break;
+        case LIST:
+            c = new ListCommand();
+            break;
+        case MARK:
+            int index = Parser.parseMarkOrDeleteCommands(userInput);
+            c = new MarkCommand(index);
+            break;
+        case UNMARK:
+            index = Parser.parseMarkOrDeleteCommands(userInput);
+            c = new UnmarkCommand(index);
+            break;
+        case DELETE:
+            index = Parser.parseMarkOrDeleteCommands(userInput);
+            c = new DeleteCommand(index);
+            break;
+        case TODO:
+            c = new ToDoCommand(Parser.parseTodoCommand(userInput));
+            break;
+        case DEADLINE:
+            // parsed[0] is description of task; parsed[1] is by
+            String[] parsed = Parser.parseDeadlineCommand(userInput);
+            c = new DeadlineCommand(parsed[0], parsed[1]);
+            break;
+        case EVENT:
+            parsed = Parser.parseEventCommand(userInput);
+            c = new EventCommand(parsed[0], parsed[1], parsed[2]);
+            break;
+        case FIND:
+            String toFind = Parser.parseFindCommand(userInput);
+            c = new FindCommand(toFind);
+            break;
+        default:
+        }
+        return c;
+    }
+
+    private static Commands getCommand(String userInput) throws DukeException {
+        try {
+            String strCommand = userInput.split(" ", 2)[0];
+            Commands command = Commands.valueOf(strCommand.toUpperCase());
+            return command;
+        } catch (IllegalArgumentException e) {
+            throw new DukeException("Invalid command");
+        } catch (NullPointerException e) {
+            throw new DukeException("Null exception encountered");
+        }
+    }
+
+    private static int parseMarkOrDeleteCommands(String userInput) throws DukeException {
+        String[] splitStr = userInput.split(" ", 2);
+        if (splitStr.length < 2) {
+            throw new DukeException("Mark / Unmark / Delete commands require an "
+                    + "integer argument referring to task number");
+        } else {
+            try {
+                int taskNumber = Integer.parseInt(splitStr[1]);
+                return taskNumber;
+            } catch (NumberFormatException e) {
+                throw new DukeException("Format of argument cannot be parsed as an integer");
+            }
+        }
+    }
+
+    private static String parseTodoCommand(String userInput) throws DukeException {
+        String[] splitStr = userInput.split(" ", 2);
```

Comment:
Maybe this could be named strSplit instead of splitStr?

This is noticed in some other parts of the code as well.